src/libcollections/vec.rs

```diff
@@ -1106,12 +1106,36 @@ impl<T: Clone> Vec<T> {
         let len = self.len();
 
         if new_len > len {
-            self.extend(repeat(value).take(new_len - len));
+            self.extend_elem(new_len - len, value);
```

Comment:
It seems like a comment stating that this was measured to be the most performant way when the code was written is useful, so that it is clear that that was the only criterion for chosing it over `extend`/`repeat` (i.e. someone reading the code if/when `extend`/`repeat` is the fastest, isn't left wondering about whether there was some other reason).